### DISCUSSION TITLE
Extension: Webhooks

### DISCUSSION THREAD

**Discussion by dawidurbanski** [wpappointments/wpappointments#56] [Ideas]
Allow sending outgoing webhooks when (not limited to):
- appointment created
- appointment updated
- appointment cancelled
- appointment confirmed
- appointment no show
- customer created
- customer updated
- customer deleted
- etc...

Nice to have:

Option to set incoming webhook handler from other systems (for example to add appointment if someone added entry in the external CRM system)

**Comment by ushanbb** [comment]
This is a great feature. 
would like to include Reminder trigger if possible 🙏 ao that a reminder can be sent to sms or web push notification for the client. The most common reason appointments fail is customers forgetting then cancelling. It’s going to be important to be able to utilise a reminder. 
thank you!

**Comment by wppoland** [comment]
Good idea. Webhooks will support all appointment lifecycle events, so you could wire up reminders via SMS or push notification services on your end. We will keep reminder triggers in mind when designing the webhook payload structure.
